qiskit_ibm_runtime/session.py

```diff
@@ -137,15 +139,33 @@ def run(
             inputs: Program input parameters. These input values are passed
                 to the runtime program.
             options: Runtime options that control the execution environment.
-                See :class:`qiskit_ibm_runtime.RuntimeOptions` for all available options.
+                See :class:`qiskit_ibm_runtime.RuntimeOptions` for all available options,
+                EXCEPT ``backend``, which should be specified during session initialization.
             callback: Callback function to be invoked for any interim results and final result.
 
         Returns:
             Submitted job.
+
+        Raises:
+            IBMInputValueError: If a backend is passed in through options that does not match
+                the current session backend.
         """
 
         options = options or {}
-        options["backend"] = self._backend
+        if "backend" in options:
+            issue_deprecation_msg(
+                "'backend' is no longer a supported option within a session",
+                "0.9",
+                "Instead, specify a backend when creating a Session instance.",
+                3,
+            )
+            if self._backend and options["backend"] != self._backend:
+                raise IBMInputValueError(
```

Comment:
There is no test coverage for this is there?

Reply:
added test case here https://github.com/Qiskit/qiskit-ibm-runtime/pull/763